### PR TITLE
Tb fix branch

### PR DESCRIPTION
**Pull request by TouchBoarder**
-- Fixed nullpointer when receiving data on new image from camera, we instead fetch the image from the uri where we wanted the image to be stored.

-- Fixed nullpointer on setLikesStatus mLikeIcon, if null we check if we can still findViewById and if it's not null the 'like drawable' can be set.

### CODE REVIEW

**Comment by nicolasgarnier** [comment]
@gkaldevrel please review!

**Comment by TouchBoarder** [comment]
status?

File: android/app/src/main/java/com/google/firebase/samples/apps/friendlypix/NewPostActivity.java [hunk 1]
```diff
@@ -196,18 +196,26 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
...
             if (resultCode == Activity.RESULT_OK) {
                 final boolean isCamera;
-                if (data.getData() == null) {
-                    isCamera = true;
-                } else {
-                    isCamera = MediaStore.ACTION_IMAGE_CAPTURE.equals(data.getAction());
-                }
-                if (!isCamera) {
-                    mFileUri = data.getData();
+                if(data!=null) {
```
Comment (gkaldevrel) [inline android/app/src/main/java/com/google/firebase/samples/apps/friendlypix/NewPostActivity.java]: Please add space between keyword and opening parenthesis, and spaces around "!=".

File: android/app/src/main/java/com/google/firebase/samples/apps/friendlypix/NewPostActivity.java [hunk 2]
```diff
@@ -196,18 +196,26 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
...
+                    if (data.getData() == null) {
+                        isCamera = true;
+                    } else {
+                        isCamera = MediaStore.ACTION_IMAGE_CAPTURE.equals(data.getAction());
+                    }
+                    if (!isCamera) {
+                        mFileUri = data.getData();
+                    }
+
+                }else{
```
Comment (gkaldevrel) [inline android/app/src/main/java/com/google/firebase/samples/apps/friendlypix/NewPostActivity.java]: Please add spaces around 'else' keyword.

File: android/app/src/main/java/com/google/firebase/samples/apps/friendlypix/NewPostActivity.java [hunk 3]
```diff
@@ -196,18 +196,26 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
...
+                    Log.d(TAG, "No Received file uri!, uses the file location");
+//                    Toast.makeText(this,"No data on result",Toast.LENGTH_SHORT).show();
```
Comment (gkaldevrel) [inline android/app/src/main/java/com/google/firebase/samples/apps/friendlypix/NewPostActivity.java]: Please delete commented-out code.

File: android/app/src/main/java/com/google/firebase/samples/apps/friendlypix/PostViewHolder.java [hunk 4]
```diff
@@ -141,13 +142,18 @@ public void setPostClickListener(PostClickListener listener) {
     }
 
     public void setLikeStatus(LikeStatus status, Context context) {
-        mLikeIcon.setImageDrawable(ContextCompat.getDrawable(context,
-                status == LikeStatus.LIKED ? R.drawable.heart_full : R.drawable.heart_empty));
+//        Check if we still have a reference to the ImageView
+        if (mLikeIcon == null && mView != null)
```
Comment (gkaldevrel) [inline android/app/src/main/java/com/google/firebase/samples/apps/friendlypix/PostViewHolder.java]: Please add braces surrounding the 'if' block body according to the style guide: https://source.android.com/source/code-style.html#use-standard-brace-style

**Comment by gkaldevrel** [comment]
Sorry for the delay -- I lost track of this. Please tweak a little to follow our coding style and I will merge it. Thanks for your contribution!

**Comment by samtstern** [comment]
Since `friendlypix-android` has been deprecated for a while I am going to close this issue.

**Pull request closed without merging**